lib/ansible/modules/windows/win_certificate.ps1

```diff
@@ -0,0 +1,159 @@
+
+#check if certificate file is available and gather details about it like type, thumbprint, subject
+if (Test-Path -Path $path) {
+    $certExtension = ([IO.Path]::GetExtension($path)).ToLower()
+
+    #proceed according to the file extension
+    if ($certExtension -eq ".pfx" -or $certExtension -eq ".p12") {
+        try {
+            $certObj = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2($path,(ConvertTo-SecureString -String $password -AsPlainText -Force))
+        }
+        catch {
+            #unable to load the certificate object, either the certificate is damaged or the password is incorrect
+            Fail-Json $result "Unable to gather certificate properties. Either the provided password is incorrect or the certificate file is corrupted. exception: $($error[0].Exception.Message)"
+        }
+    } elseif ($certExtension -eq ".cer" -or $certExtension -eq ".crt" -or $certExtension -eq ".p7b" -or $certExtension -eq ".spc") {
+        try {
+            $certObj = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2($path)
+        }
+        catch {
+            #unable to load the certificate object, either the certificate is damaged or the password is incorrect
+            Fail-Json $result "Unable to gather certificate properties. exception: $($error[0].Exception.Message)"
+        }
+        
+    } else {
+        #unkown or unsupported file extension
+        Fail-Json $result "Unkown or unsupported certificate file extension. extension: $($certExtension)"
+    }
+    $certThumbprint = $certObj.Thumbprint
+    $certSubject = $certObj.Subject
+    $certHasPK = $certObj.HasPrivateKey
+	$certNotValidBefore = ConvertTo-UnixDate($certObj.NotBefore)
+	$certNotValidAfter = ConvertTo-UnixDate($certObj.NotAfter)
+} else {
+    #certificate file not found
+    Fail-Json $result "Certificate file not found. path: $($path)"
+}
+
+#update result object with gathered cert facts
+$result.thumbprint = $certThumbprint
+$result.subject = $certSubject
+$result.hasPrivateKey = $certHasPK
+$result.notValidBefore = $certNotValidBefore
+$result.notValidAfter = $certNotValidAfter
+
+#choose action depending on desired state
+switch ($state) {
+    "present" {
+        #check if certificate is already imported
+        if (Test-Path -Path "Cert:\LocalMachine\$($store)\$($certThumbprint)") {
+            if (!$force) {
+                #the certificate is already present and we dont want to reimport (force = false)
+                $result.changed = $false
+                $result.msg = "Certificate is already present."
+                Exit-Json $result
+            }
+        }
+        if ($certExtension -eq ".pfx" -or $certExtension -eq ".p12") {
+            try {
+                #import the certificate to the desired store
+                if (!$check_mode) { $returnObj = Import-PfxCertificate -Exportable:$exportable -Password (ConvertTo-SecureString -String $password -AsPlainText -Force) `
```

Comment:
Please can you check if Import-PfxCertificate is available in powershell 3/ WMF 3.0.  
If not please document which versions of Powershell/ WMF this module will work in in the 'Requires' section of the documentation.